src/index.ts

```diff
@@ -39,10 +49,16 @@ function main(): void {
 
   registerExternalValues(tracing);
   const logger = container.resolve<Logger>(Services.LOGGER);
-  void run(logger).catch((err) => {
-    const error = err as Error;
-    logger.error(error.message);
-  });
+  setInterval(() => {
```

Comment:
Instead of set interval, that will crash on single error, consider using this methodology:
`
const intervalMs = config.get<number>('intervalMs');

while (true) {
    try {
      await run(logger);
      await new Promise((resolve) => setTimeout(resolve, intervalMs));
    } catch (error) {
      // log error here
      await new Promise((resolve) => setTimeout(resolve, intervalMs));
    }
  }
`